src/pages/index.js

```diff
@@ -112,51 +118,223 @@ const modalPreviewCloseButton = document.querySelector("#modalPreviewClose");
 //   document.removeEventListener("click", handleOverlayClose);
 // }
 
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "23172f33-55e2-4e0e-a695-0bae3ab40106",
+});
+
+let cardSection;
+
+api.getInitialCards().then((cards) => {
+  cardSection = new Section({ items: cards }, renderCard, "#section");
+
+  function renderCard(cards) {
+    const cardElement = new Card(
+      cards,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleCardLike,
+      handleCardLikeRemove
+    );
+    console.log(cards);
+    cardSection.addItem(cardElement.getView());
+  }
+
+  cardSection.renderItems();
+});
+
 const newCardPopup = new PopupWithForm({
   popupSelector: "#contentModal",
   handleFormSubmit: handleCardFormSubmit,
 });
 
+contentAddButton.addEventListener("click", () => {
+  cardFormValidator.toggleButtonState();
+  newCardPopup.openModal();
+});
+
+function handleCardFormSubmit(cardValues) {
+  const submitButton = document.getElementById("modalContentSubmit");
+  const originalButtonText = submitButton.textContent;
+  // Change button text to "Saving..."
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardValues)
+    .then((cardData) => {
+      const newCard = new Card(
+        cardData,
+        "#card-template",
+        handleImageClick,
+        handleCardDeleteClick,
+        handleCardLike,
+        handleCardLikeRemove
+      );
+      cardSection.prependItems(newCard.getView());
+      submitButton.textContent = originalButtonText;
+      newCardPopup.closeModal();
+    })
+    .catch((err) => {
+      console.log(err);
+      submitButton.textContent = originalButtonText;
+    });
+}
+
+const newUserInfo = new UserInfo(
+  "#profilename",
+  "#profilejob",
+  "#profileImage"
+);
+
+api.loadUserInfo().then((userData) => {
+  newUserInfo.setUserInfo({
+    userName: userData.name,
+    userDescription: userData.about,
+    userAvatar: userData.avatar,
+  });
+});
+
+editProfileButton.addEventListener("click", () => {
+  nameInput.value = profileName.textContent;
+  jobInput.value = profileJob.textContent;
+  cardFormValidator.toggleButtonState();
+  profileCardPopup.openModal();
+  profileCardPopup.setSubmitAction(() => {
+    const name = nameInput.value;
+    const about = jobInput.value;
+    const submitButton = document.getElementById("modalProfileSubmit");
+
+    const originalButtonText = submitButton.textContent;
+    // Change button text to "Saving..."
+    submitButton.textContent = "Saving...";
+    api
+      .editProfileInfo({
+        name: name,
+        about: about,
+      })
+      .then((submitButton.textContent = originalButtonText))
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 

**Please, correct it everywhere**